### ISSUE TITLE
ChunkStore read vs. VersionStore read (

### ISSUE THREAD

**Issue by tschm** [manahl/arctic#605]
#### Arctic Version
The latest greatest

#### Arctic Store
ChunkStore 

#### Description of problem and/or code sample that reproduces the issue

Sorry, I shall admit that I have spent only a few hours with arctic and I think it's worth digging a lot deeper. Maybe someone can help me with a questions:

ChunkStore read and VersionStore read return rather different objects. Can you please shed some light on your design here?

Also ChunkStore update fails is the symbol doesn't exist yet. Could you please add an update_or_write? 

It was quite messy getting the gcc compiler into (and out again of) my container. Could you please distribute arctic via anaconda, e.g. local compiling shouldn't be necassary. 

In your template for issues you may want to describe how to get the arctic version

If I don't specify the Chunker in the write for Chunkstore it will use the DateChunker? Shouldn't that be the PassThroughChunker as I don't want any chunking going on here? I use only daily data as in the glory old days of the CTA industry ;-)

**Comment by bmoscon** [comment]
versionstore returns a versioned object because there are potentially lots of versions of the data. It also has other information in the versionedItem object. ChunkStore's read just returns the data. 

update has an optional parameter, `upsert` that lets you write if the symbol does not exist. 

if you install that latest version, 1.68.0 (has to be currently installed from source since it hasnt been released yet) there is no need to build anything.

arctic version is easily found via pip freeze or by looking at setup.py (if from source) or by looking at egg/wheel you install from pypi. 

PassThroughChunker is for a very specific use case - writing data you do not want chunked. DailyData is a great usecase for chunkstore, or version store.

**Comment by bmoscon** [comment]
you might want to read over the docs here:

https://github.com/manahl/arctic/blob/master/docs/README.md

only the overwiew and chunkstore are done, version store is still being written.

**Comment by tschm** [comment]
ok, many thanks. 
Just a quick remark on the gcc issue. I install via
git+https://github.com/manahl/arctic.git
in my requirements.txt

This should give me the latest greatest source?

Here are the first few lines of my Dockerfile

```
FROM continuumio/miniconda3 as builder

COPY requirements.txt requirements.txt

RUN pip install --upgrade pip && \
    conda update -y conda && \
    apt-get -y update && \
    apt-get install -y postgresql-client && \
    #apt-get install -y gcc && \
    conda install -y nomkl pandas=0.23.1 requests xlrd xlsxwriter && \
    pip install --no-cache-dir  -r requirements.txt && rm requirements.txt && \
    conda clean -y --all 
    # && \
    #apt-get remove -y gcc
```
I have commented the gcc install (and subsequent delete). This is currently not building as arctic needs gcc. My requirements.txt looks like

psycopg2-binary
sqlalchemy==1.2.8
influxdb==5.1.0
git+https://github.com/manahl/arctic.git

It all works fine once gcc is installed.

**Comment by bmoscon** [comment]
arctic itself no longer needs gcc. A package it requires might as it uses many packages like numpy, pandas, etc that have C code in them.

**Comment by tschm** [comment]
yes, very correct hint. It was the lz4 package. I am installing it now with conda before I then install arctic with pip. All works without gcc. Many thanks

Now back to the chunking business. If I store only one datapoint per day (let's call this Wintan mode). Why should I chunk at all? Is there any point of chunking say a series of only 2500 points (approx 10 years)? 

Also, let's say I store OPEN, HIGH, LOW, CLOSE (PX_...) for like 1000 symbols. Shall I use 4 libraries? and 1000 symbols in each library? Could I have libraries in a library? What about chunking? Shall I use one library with 4 symbols? (I somewhere read I could read columns from a serialized DataFrame).  

I fill the database by looping over all symbols and extract symbol by symbol. I would like to avoid to extract a frame, update a frame in my main loop and then write the frame back.

**Comment by bmoscon** [comment]
it depends on how "big" that datapoint is. It also depends on how you want to read the data. If you're always reading all of it back, no need. This sounds like a good candidate for VersionStore. 

I'd think that 1000 symbols with OHLC+ would be a single dataframe. The index would be two columns - date and symbol, and O,H,L,C, etc would each be columns in the df. This could easily be stored in a single arctic symbol. I'd recommend chunking monthly (not daily). You can use `update` to update specific rows without having to read back the data (in chunkstore).
